### PR TITLE
✨ feat: add patterns listing in the graph sidebar

### PR DESCRIPTION
**Pull request by AntonyDerache**
# Pull request for Candy-board

## Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../../CONTRIBUTING.md) to
this repository.

- [X] Make sure you are requesting to **pull a feature/bugfix branch** (right
  side). Don't request your master!
- [X] Make sure you are making a pull request against the **main branch** (left
  side).
- [X] Check the commit's or even all commits' message styles matches our
  requested structure.
- [X] Check your code additions will fail neither code linting checks nor unit
  test.

## Description

All patterns of the json are listed in the graph's page sidebar.

In the pattern section you can:
- center the graph on a specific pattern
- Hide the non-related nodes of a specific pattern
- filter the list to search for a pattern
- put the tools and legend tab as sticky to keep it open while navigating inside the pattern list

❤️ Thank you!

### CODE REVIEW

File: src/components/GraphSidebar/ElementContainer.svelte [hunk 1]
```diff
@@ -3,12 +3,16 @@
   import type { IconDefinition } from "@fortawesome/free-solid-svg-icons";
   import { createEventDispatcher } from "svelte";
 
+  import type { IconOptionsDefinition } from "./types";
+
   export let label: string;
   export let icon: IconDefinition;
+  export let iconOptions: Array<IconOptionsDefinition> | null = null;
```
Comment (Yugo-Cartron) [inline src/components/GraphSidebar/ElementContainer.svelte:10]: Maybe it's better to set it to
`export let iconOptions: Array<IconOptionsDefinition> | undefined;`

File: src/components/GraphSidebar/Elements/Patterns/Pattern.svelte [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      newPatternId: patternId,
+      currentPatternId: currentNodeIdHideSelected,
+    };
+    currentNodeIdHideSelected = currentNodeIdHideSelected === patternId ? undefined : patternId;
+    dispatch("hideNodeNeighbors", args);
+  };
+</script>
+
+{#if pattern.children?.length > 0}
+  <!-- accordion -->
```
Comment (Yugo-Cartron) [inline src/components/GraphSidebar/Elements/Patterns/Pattern.svelte:29]: Suppress the comment

File: src/components/GraphSidebar/Elements/Patterns/Pattern.svelte [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+      <button slot="options" on:click|stopPropagation="{(e) => hideNodeNeighbors(e, pattern.id)}">
+        <Fa icon="{currentNodeIdHideSelected === pattern.id ? faEyeSlash : faEye}" />
+      </button>
+      {#each pattern.children as childPattern}
+        <svelte:self pattern="{childPattern}" on:centerNode on:hideNodeNeighbors />
+      {/each}
+    </candy-accordion>
+  </div>
+{:else}
+  <!-- sidebar-element -->
```
Comment (Yugo-Cartron) [inline src/components/GraphSidebar/Elements/Patterns/Pattern.svelte:44]: Same here, the accordion and the sidebar element tags are well named, it's useless to add those comments

File: src/components/GraphSidebar/Elements/Patterns/Pattern.svelte [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+  <div class="{`flex ${pattern.classes}`}">
+    <candy-sidebar-element class="w-full" label="{`${pattern.name}`}" minimizeOptions="{true}">
+      <button slot="options" on:click|stopPropagation="{() => centerCameraOnNode(pattern.id)}">
```
Comment (Yugo-Cartron) [inline src/components/GraphSidebar/Elements/Patterns/Pattern.svelte:47]: Maybe I confuse with Vue.js, but I guess you can write : 
```suggestion
      <button slot="options" on:click|stopPropagation="{centerCameraOnNode(pattern.id)}">
```

File: src/components/GraphSidebar/Elements/Patterns/PatternsList.svelte [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+<div class="patterns-container">
+  <div class="pb-1">
+    <candy-input
+      on:input="{handleInputChange}"
+      value="{inputText}"
+      label=""
+      shortcut=""
+      placeholder="Search pattern..."></candy-input>
+  </div>
+  {#if filteredPatterns === undefined || filteredPatterns.length === 0}
```
Comment (Yugo-Cartron) [inline src/components/GraphSidebar/Elements/Patterns/PatternsList.svelte:52]:
```suggestion
  {#if filteredPatterns || filteredPatterns.length === 0}
```

File: src/pages/DemoCytoscape.svelte [hunk 6]
```diff
@@ -6,17 +6,15 @@
   import expandCollapse from "cytoscape-expand-collapse";
   import popper from "cytoscape-popper";
 
-  import { makeTippy } from "@Src/tools/Cytoscape/Tippy";
-  import webCommerceContextsJson from "@Src/tools/Cytoscape/cytoscapeJson/web-commerce-contexts.json";
-  import Style from "@Src/tools/Cytoscape/Style";
-  import type { NodesTippy } from "@Src/tools/Cytoscape/Events";
-  import { setCytoscapeEvents, setHideNeighborsValue } from "@Src/tools/Cytoscape/Events";
+  import { makeTippy } from "@Src/tools/Cytoscape/tippy";
```
Comment (Yugo-Cartron) [inline src/pages/DemoCytoscape.svelte:9]: Why did you remove Uppercase ?

**Review by Yugo-Cartron — APPROVED** [review]

File: src/components/GraphSidebar/ElementContainer.svelte (hunk 1 above)
Comment (AntonyDerache) [reply src/components/GraphSidebar/ElementContainer.svelte:10]: done

File: src/components/GraphSidebar/Elements/Patterns/Pattern.svelte (hunk 2 above)
Comment (AntonyDerache) [reply src/components/GraphSidebar/Elements/Patterns/Pattern.svelte:29]: done

File: src/components/GraphSidebar/Elements/Patterns/Pattern.svelte (hunk 3 above)
Comment (AntonyDerache) [reply src/components/GraphSidebar/Elements/Patterns/Pattern.svelte:44]: done

File: src/components/GraphSidebar/Elements/Patterns/Pattern.svelte (hunk 4 above)
Comment (AntonyDerache) [reply src/components/GraphSidebar/Elements/Patterns/Pattern.svelte:47]: I can't because I am giving a parameter. It would work in this case: `on:click|stopPropagation="{centerCameraOnNode}"`

File: src/components/GraphSidebar/Elements/Patterns/PatternsList.svelte (hunk 5 above)
Comment (AntonyDerache) [reply src/components/GraphSidebar/Elements/Patterns/PatternsList.svelte:52]: Done

File: src/pages/DemoCytoscape.svelte (hunk 6 above)
Comment (AntonyDerache) [reply src/pages/DemoCytoscape.svelte:9]: Because it is a function
